### PR TITLE
JDBC

### PR DESCRIPTION
**Pull request by Bsokhatiuk**
JDBC task

### CODE REVIEW

File: BogdanSokhatiuk/pom.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        <artifactId>ACP15</artifactId>
+        <groupId>ua.artcode</groupId>
+        <version>1.0</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>Bsokhatiuk</artifactId>
+
+
+    <build>
```
Comment (presly808) [inline BogdanSokhatiuk/pom.xml:13]: remove from module, use parent configuration

File: BogdanSokhatiuk/pom.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        <!-- https://mvnrepository.com/artifact/junit/junit -->
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>5.1.6</version>
```
Comment (presly808) [inline BogdanSokhatiuk/pom.xml]: extract version to a properties

File: BogdanSokhatiuk/src/main/java/jdbcmySql/Group.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package jdbcmySql;
+
+/**
+ * Created by lost on 12.11.2016.
+ */
+public class Group {
+    private int id;
+    private String name;
+
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/Group.java]: redunant empty lines

File: BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcBushIpml.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        return resultSet;
+    }
+
+    @Override
+    public boolean sqlDeleteUpdateInsert(String sql) throws SQLException {
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcBushIpml.java]: use Log4J

File: BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcTask.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by lost on 12.11.2016.
+ */
+public class JdbcTask<T> {
+    T t;
+
+    public static final String SQLSELECT1 = "Select * from students";
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcTask.java]: extract constants to a separate class

File: BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcTask.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+    public List<T> getAll(Class cl) {
+        if (cl.equals(Student.class)) return getAllfromBase(cl, SQLSELECT1);
+        if (cl.equals(Group.class)) return getAllfromBase(cl, SQLSELECT2);
+        if (cl.equals(Subject.class)) return getAllfromBase(cl, SQLSELECT3);
+        if (cl.equals(Teacher.class)) return getAllfromBase(cl, SQLSELECT4);
+        else return null;
+    }
+
+    private List<T> getAllfromBase(Class cl, String sql) {
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcTask.java]: simplify method, split by little method

File: BogdanSokhatiuk/src/main/java/jdbcmySql/SqlConnection.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    private Logger logger;
+    private Connection connection;
+
+    public SqlConnection() {
+        url = PropertiesHolder.getProperty("URL");
+        user = PropertiesHolder.getProperty("USER");
+        pass = PropertiesHolder.getProperty("PASS");
+    }
+    @Override
+    public Connection getConnection() {
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/SqlConnection.java]: change to static method, SqlConnection.getConnection()

File: BogdanSokhatiuk/src/main/java/jdbcmySql/Student.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
+package jdbcmySql;
+
+/**
+ * Created by lost on 12.11.2016.
+ */
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/Student.java]: remove uneccessary empty lines

File: BogdanSokhatiuk/src/main/java/jdbcmySql/Subject.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public Subject(int id, String name, String description) {
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/Subject.java]: group constructors together

File: BogdanSokhatiuk/src/main/resources/db_property.properties [hunk 10]
```diff
@@ -0,0 +1,3 @@
+PASS=18199309
+USER=root
+URL=jdbc:mysql://localhost:3306/testDB
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/resources/db_property.properties]: use h2 in memory database for tests

File: BogdanSokhatiuk/src/test/java/jdbcmySql/testSqlSelect.java [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+    }
+
+
+    @Test
+    public void testInsert() {
+        String name = "";
+        Statement statement = null;
+        ResultSet resultSet = null;
+        try (PreparedStatement preparedStatement = connection.prepareStatement("INSERT INTO students VALUES(?,?,?)")) {
```
Comment (presly808) [inline BogdanSokhatiuk/src/test/java/jdbcmySql/testSqlSelect.java]: dont use low-level logic, use dao pattern

**Review by presly808 — COMMENTED** [review]
consired comments

File: BogdanSokhatiuk/pom.xml (hunk 1 above)
Comment (Bsokhatiuk) [reply BogdanSokhatiuk/pom.xml:13]: done

File: BogdanSokhatiuk/pom.xml (hunk 2 above)
Comment (Bsokhatiuk) [reply BogdanSokhatiuk/pom.xml]: done

File: BogdanSokhatiuk/src/main/java/jdbcmySql/Group.java (hunk 3 above)
Comment (Bsokhatiuk) [reply BogdanSokhatiuk/src/main/java/jdbcmySql/Group.java]: done

File: BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcBushIpml.java (hunk 4 above)
Comment (Bsokhatiuk) [reply BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcBushIpml.java]: done

File: BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcTask.java (hunk 5 above)
Comment (Bsokhatiuk) [reply BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcTask.java]: done

File: BogdanSokhatiuk/src/main/resources/db_property.properties (hunk 10 above)
Comment (Bsokhatiuk) [reply BogdanSokhatiuk/src/main/resources/db_property.properties]: done

File: BogdanSokhatiuk/src/main/java/jdbcmySql/Subject.java (hunk 9 above)
Comment (Bsokhatiuk) [reply BogdanSokhatiuk/src/main/java/jdbcmySql/Subject.java]: done

File: BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcBushIpml.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Override
+    public ResultSet sqlSelect(String sql) throws SQLException {
+        ResultSet resultSet = null;
+        try  {
+            Statement statement = connection.createStatement();
+            resultSet = statement.executeQuery(sql);
+        } catch (SQLException e) {
+            e.printStackTrace();}
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/JdbcBushIpml.java]: where is the logger?

File: BogdanSokhatiuk/src/main/java/jdbcmySql/dbMySql/SqlConnection.java [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public Connection getConnection() {
+
+        try {
+
+            connection = DriverManager.getConnection(url, user, pass);
+            LOG.debug("Create data base connection " + "Url=" + url + "; User=" + user);
+        } catch (SQLException e) {
+            LOG.error(e.getMessage());
+            e.printStackTrace();
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/dbMySql/SqlConnection.java]: if you work with a logger, dont use ```e.printStackTrace();```

File: BogdanSokhatiuk/src/main/java/jdbcmySql/h2db/Myh2Db.java [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+import java.sql.SQLException;
+import java.sql.Statement;
+
+
+
+/**
+ * Created by lost on 20.11.2016.
+ */
+public class Myh2Db {
+    private static final String DB_DRIVER = "org.h2.Driver";
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/h2db/Myh2Db.java]: get all db values from properties file

File: BogdanSokhatiuk/src/main/java/jdbcmySql/h2db/Myh2Db.java [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+            while (rs.next()) {
+                System.out.println("Id "+rs.getInt("id")+" Name "+rs.getString("name"));
+            }
+            selectPreparedStatement.close();
+
+            connection.commit();
+        } catch (SQLException e) {
+            System.out.println("Exception Message " + e.getLocalizedMessage());
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/h2db/Myh2Db.java]: where is a logger?

File: BogdanSokhatiuk/src/main/java/jdbcmySql/h2db/Myh2Db.java [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+            ResultSet rs = selectPreparedStatement.executeQuery();
+            System.out.println("H2 Database inserted through PreparedStatement");
+            while (rs.next()) {
+                System.out.println("Id "+rs.getInt("id")+" Name "+rs.getString("name"));
+            }
+            selectPreparedStatement.close();
+
+            connection.commit();
+        } catch (SQLException e) {
+            System.out.println("Exception Message " + e.getLocalizedMessage());
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/h2db/Myh2Db.java]: use logger

File: BogdanSokhatiuk/src/main/java/jdbcmySql/model/Constants.java [hunk 17]
```diff
@@ -0,0 +1,17 @@
+package jdbcmySql.model;
+
+/**
+ * Created by lost on 19.11.2016.
+ */
+public class Constants {
+    public static final String SQLSELECT1 = "Select * from students";
```
Comment (presly808) [inline BogdanSokhatiuk/src/main/java/jdbcmySql/model/Constants.java]: set appropriate and clear names to constants

i recommend to group queries by a class name(model)

File: BogdanSokhatiuk/src/test/java/jdbcmySql/testSqlSelect.java [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.sql.*;
+
+/**
+ * Created by lost on 12.11.2016.
+ */
+public class TestSqlSelect {
+    Connection connection;
```
Comment (presly808) [inline BogdanSokhatiuk/src/test/java/jdbcmySql/testSqlSelect.java]: where are access modifiers

File: BogdanSokhatiuk/src/test/java/jdbcmySql/testSqlSelect.java [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+        try (Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery("SELECT name FROM students where id=11")) {
+
+
+            while (resultSet.next()) {
+                name = resultSet.getString(1);
+            }
+            Assert.assertEquals(name, "Bogdan");
+        } catch (SQLException e) {
+            e.printStackTrace();
```
Comment (presly808) [inline BogdanSokhatiuk/src/test/java/jdbcmySql/testSqlSelect.java]: use logger

**Review by presly808 — COMMENTED** [review]
see all comments and create a dao layer

**Pull request merged**